specs/agents/tracing-instrumentation-aws.md

```diff
@@ -43,17 +43,27 @@ The format should be `DynamoDB <ActionName> <TableName>`. So for example, `Dynam
 
 ### SQS (Simple Queue Service)
 
-AWS Simple Queue Service is a message queuing service. The [messaging spec](tracing-instrumentation-messaging.md) can 
+AWS Simple Queue Service is a message queuing service. The [messaging spec](tracing-instrumentation-messaging.md) can
 be used for instrumenting SQS, but the follow specifications supersede those of the messaging spec.
 
 - **`context.destination.cloud.region`**: mandatory. The AWS region where the queue is.
 
+#### Distributed Tracing
+
 For distributed tracing, the SQS API has "message attributes" that can be used in lieu of headers.
 
+Agents should use an attribute name of `Traceparent` or `Elastic-Trace-Parent` when sending the traceparent headers via the SQS message attributes.  Agents should use an attribute name of `Tracecontext` or `Elastic-Trace-Context` if sending tracecontext headers in an SQS message attribute.
+
+Agents should default to using the non-namespaced `Traceparent` and `Tracecontext` versions of the attribute names.  Agents should use the namespaced versions of the attribute names only if the `use_elastic_traceparent_header` configuration value is set to `true`.
```

Comment:
We've introduced `use_elastic_traceparent_header` to preserve compatibility with older agents that don't read the `traceparent` header yet. As this is a net-new instrumentation, and we can already use the standardized header across agents, I don't think there's a need to make this configurable. I guess we can just always use `tracecontext`. I don't see a need for the `Elastic-Trace-` prefixed headers.

Another aspect is that we should make sure that interoperability with other agents works as expected. I had a look at the OTel Java instrumentation for SQS. It seems they currently don't support the w3c headers but only the [AWS Trace Header format](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader). But I guess that's on them to also add support for w3c. 

What's a bit confusing is that they seem to both instrument the HTTP client used for the AWS SDK requests (https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/fbf7625f8902f3bca1d17c511a6460fc96d70686/instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AwsSdkInjectAdapter.java#L17) but also use message attributes for context propagation: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/fbf7625f8902f3bca1d17c511a6460fc96d70686/instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/TracingRequestHandler.java#L81.

Do you know when the HTTP client instrumentation kicks in? Is that for non-SQS-based services, maybe S3?

Reply:
Re: dropping the `Elastic-Trace-` that seems reasonable.  The counterpoint would be that a user's service/application is already using an attributed named `traceparent` for something else we'd be stomping on that value -- but this is an edge case I can live with.  If no one else objects, I'll drop that from the spec revision.

Reply:
Re: _Do you know when the HTTP client instrumentation kicks in?_ -- my java is a little rusty so I'm not sure I completely understand what's happening in the linked code, but here's some information on how the AWS SDK works. 

Amazon's SDK provides service clients for talking with their backend services.  These service clients are mostly (I want to say completely) HTTPS based.  The Amazon SDKs contain XML files that describe each service client.  When an end-user-programmer uses an Amazon SDK object to call any method (like SQS's `receiveMessage`, `sendMessage`, etc.), the SDK will reference these XML files and use them to construct an HTTPS request body from the arguments to the method call, and then POSTs those request bodies to the appropriate service URL.  The SDK will then serialize the response from the service URL and return those values. 

So I guess the short version of that is that the AWS SDK is one giant REST-ish client for all their services, and the Amazon Request object/class (`com.amazonaws.Request` in the java SDK) is the _transport layer_ for the AWS SDK.

Having said all that, I don't _think_ there's much value is propagating the trace context headers with these AWS SDK client requests.  Since these service requests are made to Amazon servers, Amazon would need to be doing something with them.  Amazon is not, to my knowledge, doing anything with them. So it makes sense to be that the headers also need to be propagated with the message attributes. 

Also -- speculating -- it look like the author of that oTel code is [associated with Amazon](https://github.com/anuraaga) in some way, so perhaps that's a sign of future plans.  Or it might just be a programmer seeing an HTTP request and saying "better add the trace context to that". 

Does that answer your question @felixbarny?

Reply:
Thanks for the context Alan! I guess AWS has some special treatment for their X-Ray headers. We should be good just adding the w3c headers for now.

Reply:
Latest version without the additional `elastic-` headers: https://github.com/elastic/apm/blob/fb5686893f2c6683d7e79950f72c4764dfc4052c/specs/agents/tracing-instrumentation-aws.md#distributed-tracing